src/Api/Serializer/JsonRpcSerializer.php

```diff
@@ -48,21 +52,41 @@ public function __construct(
      *
      * @return RequestInterface
      */
-    public function __invoke(CommandInterface $command)
+    public function __invoke(
+        CommandInterface $command,
+        $endpointProvider = null,
+        array $clientArgs = null
+    )
     {
-        $name = $command->getName();
-        $operation = $this->api->getOperation($name);
+        $operationName = $command->getName();
```

Comment:
Wanted to get your take on this.  We could encapsulate the logic for JsonRPC and Query serializers into one method.  The rest serializer would need its own.  Creating another middleware for this seemed like an unnecessary complexity— because this all needs to happen before request serialization, we'd have to devise a way to pass the resolved endpoint and headers to the next handler (the serializer) before we (historically) instantiate a request. I can write some methods to encapsulate this stuff- just didn't know if we'd want it out in the open for better readability